### DISCUSSION TITLE
surround: configure spacing per char and language?

### DISCUSSION THREAD

**Discussion by bennypowers** [nvim-mini/mini.nvim#92] [Q&A]
Hello!

Consider:

```md
I'm a markdown file
<!--  ^ cursor here -->
```

`ysiw[`

I'd prefer to receive:

```md
I'm a [markdown] file
```

But instead I'll get

```md
I'm a [ markdown ] file
```

And same for `(`

However, I might also prefer to pad my square brackets with spaces in JavaScript files

```js
const things = 1, 2, 3;
// visual:     _______
```
`S[`
```js
const things = [ 1, 2, 3 ];
```

So, can I configure mini.surround to:
1. pad or not pad, per surround char
2. specify that behaviour per filetype?

**Comment by echasnovski** [comment]
Judging by your examples, you either use 'tpope/vim-surround' or setup of 'mini.surround' providing similar experience. By default all that examples give surrounding without padding (and `S[` doesn't even work).

In [setup of 'mini.surround' similar to 'vim-surround'](https://github.com/echasnovski/mini.nvim/blob/c820abfe561a5946e1b327ff9912491876cbf366/doc/mini.txt#L4239-L4267) opening characters add pads and closing - don't. This is the most straightforward and suggested way of managing pads.

In theory, you should be able to write custom code that changes custom surroundings per filetype. Something along the lines of autocommands for specified events (`BufEnter`, `BufLeave`, `FileType`) which updates `MiniSurround.config.custom_surroundings`. However, this won't be added in module itself.

**Reply by echasnovski** [reply]
@bennypowers, setting custom surroundings is now possible with buffer local configuration. More details [here](https://github.com/echasnovski/mini.nvim/blob/ee1ea2581843b5b2a74a52f6029af3bc4a357f93/doc/mini.txt#L195-L215) and [here](https://github.com/echasnovski/mini.nvim/blob/ee1ea2581843b5b2a74a52f6029af3bc4a357f93/lua/mini/surround.lua#L266-L275).

Basically, use any suitable method for creating appropriate buffer variable `vim.b.minisurround_config` for specific filetypes. I prefer using 'after/ftplugin' directory for that, but autocommands can be used too.
